plutus-ledger-constraints/src/Ledger/Constraints/TxConstraints.hs

```diff
@@ -714,6 +717,11 @@ mustSpendPubKeyOutput = singleton . MustSpendPubKeyOutput
 -- 'Ledger.Constraints.OffChain.unspentOutputs' or through
 -- 'Ledger.Constraints.OffChain.otherData'.
 --
+-- If several calls to 'mustSpendScriptOutput' are performed for the same 'TxOutRef',
+-- if the two constraints have different redeemers, an error will be thrown when the transaction will be submitted.
```

Comment:
I think the error is not thrown when the transaction is submitted, but when the transaction is created with `mkTx`.